api/mfx_dispatch/windows/src/mfx_dispatcher.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2012-2019 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "mfx_dispatcher.h"
+#include "mfx_dispatcher_log.h"
+#include "mfx_load_dll.h"
+
+#include <assert.h>
+
+#include <string.h>
+#include <windows.h>
+#pragma warning(disable:4355)
+
+MFX_DISP_HANDLE::MFX_DISP_HANDLE(const mfxVersion requiredVersion) :
+    _mfxSession()
+    ,apiVersion(requiredVersion)
+    ,pluginHive()
+    ,pluginFactory((mfxSession)this)
+{
+    actualApiVersion.Version = 0;
+    implType = MFX_LIB_SOFTWARE;
+    impl = MFX_IMPL_SOFTWARE;
+    loadStatus = MFX_ERR_NOT_FOUND;
+    dispVersion.Major = MFX_DISPATCHER_VERSION_MAJOR;
+    dispVersion.Minor = MFX_DISPATCHER_VERSION_MINOR;
+    storageID = 0;
+    implInterface = MFX_IMPL_HARDWARE_ANY;
+
+    hModule = (mfxModuleHandle) 0;
+
+} // MFX_DISP_HANDLE::MFX_DISP_HANDLE(const mfxVersion requiredVersion)
+
+MFX_DISP_HANDLE::~MFX_DISP_HANDLE(void)
+{
+    Close();
+
+} // MFX_DISP_HANDLE::~MFX_DISP_HANDLE(void)
+
+mfxStatus MFX_DISP_HANDLE::Close(void)
+{
+    mfxStatus mfxRes;
+
+    mfxRes = UnLoadSelectedDLL();
+
+    // the library wasn't unloaded
+    if (MFX_ERR_NONE == mfxRes)
```

Comment:
Accord. to comment `MFX_ERR_NONE` means that libary was not unloaded. Quite odd as for me.